lib/slots.ts

```diff
@@ -1,137 +1,58 @@
 
-// say invitee is -60,1380, and boundary is -120,240 - the overlap is -60,240
-const getOverlaps = (inviteeBoundary: Boundary, boundaries: Boundary[]) =>
-  boundaries.map((boundary) => intersectBoundary(inviteeBoundary, boundary)).filter(Boolean);
+  console.log(thisDayWorkingHours);
 
-const organizerBoundaries = (
-  workingHours: [],
-  inviteeDate: Dayjs,
-  inviteeBounds: Boundary,
-  organizerTimeZone
-): Boundary[] => {
-  const boundaries: Boundary[] = [];
-
-  const startDay: number = +inviteeDate.startOf("d").add(inviteeBounds.lowerBound, "minutes").format("d");
-  const endDay: number = +inviteeDate.startOf("d").add(inviteeBounds.upperBound, "minutes").format("d");
-
-  workingHours.forEach((item) => {
-    const lowerBound: number = item.startTime - dayjs().tz(organizerTimeZone).utcOffset();
-    const upperBound: number = item.endTime - dayjs().tz(organizerTimeZone).utcOffset();
-    if (startDay !== endDay) {
-      if (inviteeBounds.lowerBound < 0) {
-        // lowerBound edges into the previous day
-        if (item.days.includes(startDay)) {
-          boundaries.push({ lowerBound: lowerBound - 1440, upperBound: upperBound - 1440 });
-        }
-        if (item.days.includes(endDay)) {
-          boundaries.push({ lowerBound, upperBound });
-        }
-      } else {
-        // upperBound edges into the next day
-        if (item.days.includes(endDay)) {
-          boundaries.push({ lowerBound: lowerBound + 1440, upperBound: upperBound + 1440 });
-        }
-        if (item.days.includes(startDay)) {
-          boundaries.push({ lowerBound, upperBound });
-        }
-      }
-    } else {
-      if (item.days.includes(startDay)) {
-        boundaries.push({ lowerBound, upperBound });
-      }
-    }
-  });
-
-  return boundaries;
-};
-
-const inviteeBoundary = (startTime: number, utcOffset: number, frequency: number): Boundary => {
-  const upperBound: number = freqApply(Math.floor, 1440 - utcOffset, frequency);
-  const lowerBound: number = freqApply(Math.ceil, startTime - utcOffset, frequency);
-  return {
-    lowerBound,
-    upperBound,
-  };
-};
-
-const getSlotsBetweenBoundary = (frequency: number, { lowerBound, upperBound }: Boundary) => {
   const slots: Dayjs[] = [];
-  for (let minutes = 0; lowerBound + minutes <= upperBound - frequency; minutes += frequency) {
-    slots.push(
-      dayjs
-        .utc()
-        .startOf("d")
-        .add(lowerBound + minutes, "minutes")
-    );
+  for (let minutes = getMinuteOffset(inviteeDate, frequency); minutes < 1440; minutes += frequency) {
+    const slot = inviteeDate.startOf("day").add(minutes, "minutes");
+    // add slots to available slots if it is found to be between the start and end time of the checked working hours.
+    if (
+      thisDayWorkingHours.some((hours) =>
+        slot.isBetween(
+          dayjs(inviteeDate).startOf("day").utc().add(hours.startTime, "minutes"),
+          dayjs(inviteeDate).startOf("day").utc().add(hours.endTime, "minutes"),
```

Comment:
unsure if this logic is elsewhere but i imagine it'd have  to be something like

```suggestion
          dayjs(inviteeDate).startOf("day").utc().add(hours.startTime, "minutes"),
          dayjs(inviteeDate).startOf("day").utc().add(hours.endTime, "minutes").add(hours.endTime < startTime ? 1 : 0, "day"),
```